plonky2/src/fri/oracle.rs

```diff
@@ -154,6 +154,10 @@ impl<F: RichField + Extendable<D>, C: GenericConfig<D, F = F>, const D: usize>
             alpha.shift_poly(&mut final_poly);
             final_poly += quotient;
         }
+        final_poly.trim();
+        // Multiply the final polynomial by `X`, so that `final_poly` has the maximum degree for
```

Comment:
nit: let's copy this to the two verifier changes also, otherwise I think it's hard to see why what the mul is for